commit 451eb56ab2336e92450030963b7d80bb0f973a99
Author: John Johansen <john.johansen@canonical.com>
Date:   Wed Sep 27 07:50:19 2017 -0400

    UBUNTU: SAUCE: LSM stacking: add Kconfig to set default display LSM
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/security/Kconfig b/security/Kconfig
index a3ffa29ef7f4..305b496ff6a3 100644
--- a/security/Kconfig
+++ b/security/Kconfig
@@ -377,6 +377,44 @@ config SECURITY_APPARMOR_STACKED
 
 	  If you are unsure how to answer this question, answer N.
 
+choice
+	prompt "Default LSM for legacy interfaces"
+	default SECURITY_DEFAULT_DISPLAY_SELINUX if SECURITY_SELINUX_STACKED
+	default SECURITY_DEFAULT_DISPLAY_SMACK if SECURITY_SMACK_STACKED
+	default SECURITY_DEFAULT_DISPLAY_TOMOYO if SECURITY_TOMOYO_STACKED
+	default SECURITY_DEFAULT_DISPALY_APPARMOR if SECURITY_APPARMOR_STACKED
+	default SECURITY_DEFAULT_DISPLAY_FIRST
+
+	help
+	  Select the security module context that will be displayed by
+          default on legacy interfaces if the kernel parameter
+          security.display= is not specified.
+
+	config SECURITY_DEFAULT_DISPLAY_SELINUX
+		bool "SELinux" if SECURITY_SELINUX=y
+
+	config SECURITY_DEFAULT_DISPLAY_SMACK
+		bool "Simplified Mandatory Access Control" if SECURITY_SMACK=y
+
+	config SECURITY_DEFAULT_DISPLAY_TOMOYO
+		bool "TOMOYO" if SECURITY_TOMOYO=y
+
+	config SECURITY_DEFAULT_DISPLAY_APPARMOR
+		bool "AppArmor" if SECURITY_APPARMOR=y
+
+	config SECURITY_DEFAULT_DISPLAY_FIRST
+		bool "First security module to register"
+
+endchoice
+
+config SECURITY_DEFAULT_DISPLAY_NAME
+	string
+	default "selinux" if SECURITY_DEFAULT_DISPLAY_SELINUX
+	default "smack" if SECURITY_DEFAULT_DISPLAY_SMACK
+	default "tomoyo" if SECURITY_DEFAULT_DISPLAY_TOMOYO
+	default "apparmor" if SECURITY_DEFAULT_DISPLAY_APPARMOR
+	default "" if SECURITY_DEFAULT_DISPLAY_FIRST
+
 endmenu
 
 endmenu
diff --git a/security/security.c b/security/security.c
index bb3007d19bcb..29d43dae9a8a 100644
--- a/security/security.c
+++ b/security/security.c
@@ -62,7 +62,11 @@ static __initdata char chosen_lsms[SECURITY_CHOSEN_NAMES_MAX + 1] =
 #else
 	CONFIG_DEFAULT_SECURITY;
 #endif
-static __initdata char chosen_display_lsm[SECURITY_NAME_MAX + 1];
+static __initdata char chosen_display_lsm[SECURITY_NAME_MAX + 1]
+#ifdef CONFIG_SECURITY_STACKING
+	= CONFIG_SECURITY_DEFAULT_DISPLAY_NAME
+#endif
+;
 static char default_display_lsm[SECURITY_NAME_MAX + 1];
 
 static void __init do_security_initcalls(void)
